skfp/descriptors/constitutional.py

```diff
@@ -0,0 +1,278 @@
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the atom count is to be calculated.
+
+    atom_symbol : str
+        The symbol of the atom type to count (e.g., "H" for hydrogen, "C" for carbon).
+
+    References
+    ----------
+    .. [1] `Gerta, Rucker and Christoph, Ruecker.
+        "Counts of All Walks as Atomic and Molecular Descriptors"
+        Journal of Chemical Information and Computer Sciences 33.5 (1993): 683–695.
+        <https://doi.org/10.1021/ci00015a005>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import atom_count
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> atom_count(mol, "C")
+    6
+    """
+    mol_with_h = AddHs(mol)
+    return sum(1 for atom in mol_with_h.GetAtoms() if atom.GetSymbol() == atom_symbol)
+
+
+def average_molecular_weight(mol: Mol) -> float:
+    """
+    Average Molecular Weight.
+
+    Calculates the average molecular weight of the molecule [1]_, defined as the molecular
+    weight divided by the number of atoms.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the average molecular weight is to be calculated.
+
+    References
+    ----------
+    .. [1] `Rinta, Kawagoe.
+        "Exploring Molecular Descriptors and Acquisition Functions in Bayesian
+        Optimization for Designing Molecules with Low Hole Reorganization Energy"
+        ACS Omega 9.49 (2024): 48844–48854.
+        <https://doi.org/10.1021/acsomega.4c09124>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import average_molecular_weight
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> average_molecular_weight(mol)
+    13.019
+    """
+    num_atoms = mol.GetNumAtoms()
+    if num_atoms == 0:
+        raise ValueError(
+            "The molecule has no atoms, average molecular weight cannot be calculated."
+        )
+    return MolWt(mol) / num_atoms
+
+
+def molecular_weight(mol: Mol) -> float:
+    """
+    Molecular Weight.
+
+    Calculates the molecular weight of the molecule [1]_.
+
+    Parameters
+    ----------
+    mol : RDKit Mol object
+        The molecule for which the molecular weight is to be calculated.
+
+    References
+    ----------
+    .. [1] `Hiromasa, Kaneko.
+        "Molecular Descriptors, Structure Generation,
+        and Inverse QSAR/QSPR Based on SELFIES"
+        ACS Omega 8.24 (2023): 21781–21786.
+        <https://doi.org/10.1021/acsomega.3c01332>`_
+
+    Examples
+    --------
+    >>> from rdkit.Chem import MolFromSmiles
+    >>> from skfp.descriptors.constitutional import molecular_weight
+    >>> mol = MolFromSmiles("C1=CC=CC=C1")  # Benzene
+    >>> molecular_weight(mol)
+    78.114
+    """
+    return MolWt(mol)
+
+
+def number_of_double_bonds(mol: Mol) -> int:
```

Comment:
Single, double, triple and aromatic bonds should be a single function with configurable `bond_type` parameter, with function name `bond_type_count`